packages/react/src/hooks/common.ts

```diff
@@ -170,5 +176,13 @@ export const useSignOut = (stateAll: boolean = false) => {
     (a, b) => a === b
   )
 
-  return { signOut, isSuccess }
+  const isError = !isSuccess
+
+  const error = useSelector(
+    service,
+    (state) => state.context.errors.signout || null,
+    (a, b) => a?.error === b?.error
+  )
+
+  return { signOut, isSuccess, isError, error }
```

Comment:
I think it's nice to provide both `isSuccess` and `isError` as it makes it easy to do if-statements.

It's not always the case that you can infer one state based on the other.

As an example. Just because `isSuccess` is `false` does not mean `isError` is true.

[React Query](https://react-query.tanstack.com/guides/mutations#_top) provide both boolean variables and a `status` variable:

> `isIdle` or `status === 'idle'` - The mutation is currently idle or in a fresh/reset state
> `isLoading` or `status === 'loading'` - The mutation is currently running
> `isError` or `status === 'error'` - The mutation encountered an error
> `isSuccess` or `status === 'success'` - The mutation was successful and mutation data is available

Reply:
I feel that there is rarely a use-case where we need both. Why not just simply provide the `error` object that would describe the actual error we have? The state can be "success", but we could still have an error object.

Reply:
> Why not just simply provide the error object that would describe the actual error we have?

If it will always work to do:

```js
if (error) {
  return <div>Error!</div>
}
```

Then I would be more fine removing the `isError`. 

But I think @plmercereau mentioned that you had to do `if (!!error) {` which adds extra code and complexity to developers.

Just to clarify, is your comment about removing the `isError` variable something you suggest doing for all hooks? Or just for `useSignOut`()?

Reply:
`!!error` and `error` will behave the same in such a case.

We implemented both `isError` and `isSuccess` so far, and I feel we should avoid changing our logic unless there is a clear need to do so, especially until we have a strong testing framework.

I suggest 
- we avoid implementing `isError` in `useSignOut`
- we keep it as is elsewhere
- we don't deprecate `isError`
- and we continue this discussion internally or in a separate issue

Reply:
I think we can have this discussion on a dedicated discussion channel. My personal take on things here is that we should have a really slim API wherever it is possible. We can have a more detailed discussion about this later. Let's not worry about this `isError` / `isSuccess` thing until we don't come up with a specific solution IMO.

Reply:
https://github.com/nhost/nhost/discussions/542 created a discussion